homeassistant/components/fritz/common.py

```diff
@@ -217,7 +218,17 @@ def setup(self) -> None:
         self._current_firmware = info.get("NewSoftwareVersion")
 
         self._update_available, self._latest_firmware = self._update_device_info()
-        self.device_is_router = "WANIPConn1" in self.connection.services
+        if "Layer3Forwarding1" in self.connection.services:
+            connection_type = self.connection.call_action(
+                "Layer3Forwarding1", "GetDefaultConnectionService"
+            )
+            if connection_type:
+                self.device_conn_type = connection_type["NewDefaultConnectionService"][
+                    2:
+                ][:-2]
+                self.device_is_router = self.connection.call_action(
+                    self.device_conn_type, "GetInfo"
+                ).get("NewEnable")
```

Comment:
What do you think about this approach?

```suggestion
        if "Layer3Forwarding1" in self.connection.services:
            try:
                connection_type = self.connection.call_action(
                    "Layer3Forwarding1", "GetDefaultConnectionService"
                ).get("NewDefaultConnectionService")
            except FRITZ_EXCEPTIONS as ex:
                _LOGGER.debug(
                    "could not get WAN connection type for device '%s', error: %s",
                    self.host,
                    ex,
                )
                connection_type = None

            if connection_type:
                self.device_conn_type = connection_type[2:][:-2]
                self.device_is_router = self.connection.call_action(
                    self.device_conn_type, "GetInfo"
                ).get("NewEnable")
```

Reply:
If the service is there, it must work 😉

If not I think we should know and act according, not mask; it may make a wrong assumption that device is not a router.

Reply:
OK, than just this one?

```suggestion
        if "Layer3Forwarding1" in self.connection.services:
            connection_type = self.connection.call_action(
                "Layer3Forwarding1", "GetDefaultConnectionService"
            ).get("NewDefaultConnectionService")

            if connection_type:
                self.device_conn_type = connection_type[2:][:-2]
                self.device_is_router = self.connection.call_action(
                    self.device_conn_type, "GetInfo"
                ).get("NewEnable")
```